client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,197 @@
+	} );
+}
 
 const props = {
 	callToAction: null,
 	title: 'banner title',
 };
 
 describe( 'Banner basic tests', () => {
-	test( 'should not blow up and have proper CSS class', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( '.banner' ) ).toHaveLength( 1 );
+	afterEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	test( 'should output Banner component with proper CSS class', () => {
+		const { container } = render( <Banner { ...props } /> );
+		expect( container.firstChild ).toHaveClass( 'banner' );
 	} );
 
 	test( 'should render Card if dismissPreferenceName is null', () => {
-		const comp = shallow( <Banner { ...props } dismissPreferenceName={ null } /> );
-		expect( comp.find( Card ) ).toHaveLength( 1 );
-		expect( comp.find( 'DismissibleCard' ) ).toHaveLength( 0 );
+		const { container } = render( <Banner { ...props } dismissPreferenceName={ null } /> );
+		expect( container.firstChild ).toHaveClass( 'card' );
+		expect( container.firstChild ).not.toHaveClass( 'is-dismissible' );
 	} );
 
 	test( 'should render DismissibleCard if dismissPreferenceName is defined', () => {
-		const comp = shallow( <Banner { ...props } dismissPreferenceName={ 'banner-test' } /> );
-		expect( comp.find( Card ) ).toHaveLength( 0 );
-		expect( comp.find( 'DismissibleCard' ) ).toHaveLength( 1 );
+		renderWithRedux( <Banner { ...props } dismissPreferenceName={ 'banner-test' } /> );
+		expect( screen.getByLabelText( 'Dismiss' ) ).toBeInTheDocument();
 	} );
 
 	test( 'should have .has-call-to-action class if callToAction is defined', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ 'Upgrade Now!' } /> );
-		expect( comp.find( '.has-call-to-action' ) ).toHaveLength( 1 );
+		const { container } = render( <Banner { ...props } callToAction={ 'Upgrade Now!' } /> );
+		expect( container.firstChild ).toHaveClass( 'has-call-to-action' );
 	} );
 
 	test( 'should not have .has-call-to-action class if callToAction is null', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ null } /> );
-		expect( comp.find( '.has-call-to-action' ) ).toHaveLength( 0 );
+		const { container } = render( <Banner { ...props } callToAction={ null } /> );
+		expect( container.firstChild ).not.toHaveClass( 'has-call-to-action' );
 	} );
 
 	test( 'should render a <Button /> when callToAction is specified', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ 'Buy something!' } /> );
-		expect( comp.find( Button ) ).toHaveLength( 1 );
+		render( <Banner { ...props } callToAction={ 'Buy something!' } /> );
+		expect( screen.getByRole( 'button' ) ).toHaveTextContent( 'Buy something!' );
 	} );
 
 	test( 'should not render a <Button /> when callToAction is not specified', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( Button ) ).toHaveLength( 0 );
+		render( <Banner { ...props } /> );
+		expect( screen.queryByRole( 'button' ) ).toBeNull();
```

Comment:
This does the job, but `.not.toBeInTheDocument()` seems to express our intention here much better IMHO.

```suggestion
		expect( screen.queryByRole( 'button' ) ).not. toBeInTheDocument();
```